### ISSUE TITLE
Offline Plugin does not support providing style via local files on device, only via remote URLS

### ISSUE THREAD

**Issue by pyb4430** [mapbox/mapbox-plugins-android#813]
The implementations of OfflineRegionDefinition don't have support for setting the style from a local file or something of the sort. I need to define my styles dynamically (the map sources will be selected by the user) and then cache tiles for later offline use. MapboxMap can have its style set via local file and asset urls, so why not the same support for the offline module?

**Comment by tobrun** [comment]
@pyb4430 agreed that this should be added. I added it to the backlog but feel free to provide a PR for it.

**Comment by codeversed** [comment]
@langsmith @tobrun @LukasPaczos 
I have recently stumbled upon this issue of not being able to download an offline region specifying a locally stored JSON StyleSpec. I contacted support and they confirmed the issue but linked me here for the public discussion. 

I'm a little confused about why this is an issue with the OfflinePlugin _(which we do not use)_ rather than an issue with [mapbox-gl-native-android](https://github.com/mapbox/mapbox-gl-native-android). To me, it seems like the feature would need to be added to the `OfflineManager` or `FileSource`. 

We also have an iOS app, which I confirmed this morning that they are passing a locally stored JSON StyleSpec path that has a prefix of `file://`. Is this working on iOS and not Android? The iOS SDK is a little different setup but they define a region and pass a style URL the same ask the Android SDK. 

Any insight on the root cause of the issue would be helpful, thanks!

**Comment by tobrun** [comment]
sorry for late reply @codeversed, correct that this needs to be addressed first upstream. The underlying code in offline_download.cpp uses a direct reference to the online file source and not an abstraction that would allow to use a local file or assets hosted file. This means that this shouldn't work for iOS either except if NSUrl is able to handle local files (I'm not an iOS developer so IDK).

**Comment by codeversed** [comment]
@tobrun this is a needed feature for us. I have a support ticket in but it's marked low priority. What can I do to help get this done? It currently does work on iOS.
